src/OutputReaders/field_time_series.jl

```diff
@@ -180,6 +182,34 @@ function Base.getindex(fts::FieldTimeSeries{LX, LY, LZ, OnDisk}, n::Int) where {
     return Field(loc, fts.grid; indices=fts.indices, boundary_conditions=fts.boundary_conditions, data=field_data)
 end
 
+# Linear time interpolation
+function Base.getindex(fts::FieldTimeSeries, time_index::Time)
+    Ntimes = length(fts.times)
+    time = time_index.time
+    n₁, n₂ = index_binary_search(fts.times, time, Ntimes)
+    if n₁ == n₂ # no interpolation
+        return fts[n₁]
+    end
+    
+    # fractional index
+    @inbounds n = (n₂ - n₁) / (fts.times[n₂] - fts.times[n₁]) * (time - fts.times[n₁]) + n₁
+    return compute!(Field(fts[n₂] * (n - n₁) + fts[n₁] * (n₂ - n)))
+end
+
+# Linear time interpolation
+function Base.getindex(fts::FieldTimeSeries, i::Int, j::Int, k::Int, time_index::Time)
+    Ntimes = length(fts.times)
```

Comment:
`length` is slow for big vectors right? Should we allow this to be passed into `Time`?

Reply:
then we can implement something like

```julia
maybe_diagnose_length(time::Time{<:Int}, times) = time.Ntimes
maybe_diagnose_length(::Time, times) = length(times)
```

or mahybe you have a better name